### PR TITLE
DMP-1952 - daily job list logging

### PR DESCRIPTION
**Pull request by mestebanez**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DMP-1952

### Change description ###

Introduce a logging report that enumerates a number of daily list job outcomes and logs using SL4J

Add the general ability for tests to interrogate the SL4J log framework

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/darts/dailylist/service/impl/ProcessAllDailyListsForCourthouseTest.java [hunk 1]
```diff
@@ -44,11 +45,13 @@ class ProcessAllDailyListsForCourthouseTest {
...
     @Mock
     private DailyListEntity latestDailyListEntityForLeeds;
+    @Mock
+    private LogApi logApi;
 
 
     @BeforeEach
     void setUp() {
-        dailyListProcessor = new DailyListProcessorImpl(dailyListRepository, dailyListUpdater);
+        dailyListProcessor = new DailyListProcessorImpl(dailyListRepository, dailyListUpdater, logApi);
```
Comment (davet1985) [inline src/test/java/uk/gov/hmcts/darts/dailylist/service/impl/ProcessAllDailyListsForCourthouseTest.java:54]: Can we assert that the logApi method is called appropriately.

File: src/integrationTest/java/uk/gov/hmcts/darts/dailylist/service/DailyListProcessorTest.java [hunk 2]
```diff
@@ -122,6 +200,13 @@ void dailyListProcessorMultipleDailyList() throws IOException {
 
     }
 
+    @Test
+    void dailyListProcessorNoDailyListNoReport() throws IOException {
+        dailyListProcessor.processAllDailyLists();
+
+        //Mockito.verify(logApi, times(0)).processedDailyListJob(Mockito.notNull());
```
Comment (davet1985) [inline src/integrationTest/java/uk/gov/hmcts/darts/dailylist/service/DailyListProcessorTest.java:207]: Remove comment, if it's not needed

File: src/main/java/uk/gov/hmcts/darts/dailylist/service/impl/DailyListProcessorImpl.java [hunk 3]
```diff
@@ -23,20 +26,45 @@ public class DailyListProcessorImpl implements DailyListProcessor {
...
+        int count = 0;
+        for (String entity : dailyListMap.keySet()) {
+            count = count + dailyListMap.get(entity).size();
+        }
+        return count;
+    }
+
+    private void reportStats(DailyListLogJobReport report) {
+        if (report.getTotalDailyListRegisteredJobStatus() > 0) {
+            logApi.processedDailyListJob(report);
```
Comment (davet1985) [inline src/main/java/uk/gov/hmcts/darts/dailylist/service/impl/DailyListProcessorImpl.java:64]: I think we should report the stats even if nothing was processed, that way we can see that a job was run.

**Review by davet1985 — COMMENTED** [review]
Looks good and worked when tested locally, just a few comments to consider.

File: src/test/java/uk/gov/hmcts/darts/dailylist/service/impl/ProcessAllDailyListsForCourthouseTest.java (hunk 1 above)
Comment (mestebanez) [reply src/test/java/uk/gov/hmcts/darts/dailylist/service/impl/ProcessAllDailyListsForCourthouseTest.java:54]: oh yeah though this was testing something else. The mocking of the processor at the top (plus the name of the class confused me)

Comment (mestebanez) [reply src/test/java/uk/gov/hmcts/darts/dailylist/service/impl/ProcessAllDailyListsForCourthouseTest.java:54]: Done. Changed the class name and eliminated the useless processor mock

File: src/main/java/uk/gov/hmcts/darts/log/service/impl/DailyListLogJobServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import uk.gov.hmcts.darts.log.service.DailyListLogJobService;
+import uk.gov.hmcts.darts.log.util.DailyListLogJobReport;
+
+@Service
+@AllArgsConstructor
+@Slf4j
+public class DailyListLogJobServiceImpl implements DailyListLogJobService {
+
+    @Override
+    public void logJobReport(DailyListLogJobReport report) {
```
[the remark this answers is not in the document]
Comment (mestebanez) [reply src/main/java/uk/gov/hmcts/darts/log/service/impl/DailyListLogJobServiceImpl.java:15]: Done

File: src/main/java/uk/gov/hmcts/darts/dailylist/service/impl/DailyListProcessorImpl.java (hunk 3 above)
Comment (mestebanez) [reply src/main/java/uk/gov/hmcts/darts/dailylist/service/impl/DailyListProcessorImpl.java:64]: np. Done

File: src/integrationTest/java/uk/gov/hmcts/darts/dailylist/service/DailyListProcessorTest.java (hunk 2 above)
Comment (davet1985) [reply src/integrationTest/java/uk/gov/hmcts/darts/dailylist/service/DailyListProcessorTest.java:200]: The comment is still there, it is needed?

**Review by davet1985 — APPROVED** [review]

**Pull request merged**